src/Esprima/Utils/AstJson.cs

```diff
@@ -0,0 +1,830 @@
+
+            public bool IncludeLineColumn { get; private set; }
+            public bool IncludeRange { get; private set; }
+            public LocationMembersPlacement LocationMembersPlacement { get; private set; }
+
+            public Options() {}
+
+            Options(Options options)
+            {
+                IncludeLineColumn = options.IncludeLineColumn;
+                IncludeRange = options.IncludeRange;
+                LocationMembersPlacement = options.LocationMembersPlacement;
+            }
+
+            public Options WithIncludeLineColumn(bool value) =>
+                value == IncludeLineColumn ? this : new Options(this) { IncludeLineColumn = value };
+
+            public Options WithIncludeRange(bool value) =>
+                value == IncludeRange ? this : new Options(this) { IncludeRange = value };
+
+            public Options WithLocationMembersPlacement(LocationMembersPlacement value) =>
+                value == LocationMembersPlacement ? this : new Options(this) { LocationMembersPlacement = value };
+
+            public Options WithLineColumn() => WithIncludeLineColumn(true);
+            public Options WithRange()      => WithIncludeRange(true);
+            public Options WithLocation()   => WithLineColumn().WithRange();
+
+            public Options WithLocation(LocationMembersPlacement placement) =>
+                WithLocation().WithLocationMembersPlacement(placement);
+        }
+
+        public static string ToJsonString(this INode node) =>
+            ToJsonString(node, indent: null);
+
+        public static string ToJsonString(this INode node, string indent) =>
+            ToJsonString(node, Options.Default, indent);
+
+        public static string ToJsonString(this INode node, Options options) =>
+            ToJsonString(node, options, null);
+
+        public static string ToJsonString(this INode node, Options options, string indent)
+        {
+            var writer = new StringWriter();
+            WriteJson(node, writer, options, indent);
+            return writer.ToString();
+        }
+
+        public static void WriteJson(this INode node, TextWriter writer) =>
+            WriteJson(node, writer, indent: null);
+
+        public static void WriteJson(this INode node, TextWriter writer, string indent) =>
+            WriteJson(node, writer, Options.Default, indent);
+
+        public static void WriteJson(this INode node, TextWriter writer, Options options) =>
+            WriteJson(node, writer, options, null);
+
+        public static void WriteJson(this INode node, TextWriter writer, Options options, string indent)
+        {
+            if (node == null) throw new ArgumentNullException(nameof(node));
+            if (writer == null) throw new ArgumentNullException(nameof(writer));
+            if (options == null) throw new ArgumentNullException(nameof(options));
+
+            var visitor = new Visitor(new JsonTextWriter(writer, indent),
```

Comment:
The assumption here is that the `TextWriter` ownership is with the caller so that `using` belongs somewhere up the call-stack. You see, it could be a file stream writer passed down for writing the JSON in the middle of a file. You don't want to dispose it if the caller wants to continue to write to the file. If there was another helper for writing the JSON as the content of a file then it would have the `using` like this:

```c#
public static void WriteJsonToFile(this INode node, string path, Options options, string indent)
{
    if (node == null) throw new ArgumentNullException(nameof(node));
    if (options == null) throw new ArgumentNullException(nameof(options));

    using (var writer = File.CreateText(path))
        node.WriteJson(writer, options, indent);
}
```